src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -741,10 +937,14 @@ public void ChangeConnectionDatabaseContext(string ownerUri, string newDatabaseN
             {
                 try
                 {
-                    if (info.SqlConnection.State == ConnectionState.Open)
+                    foreach (DbConnection connection in info.GetAllConnections())
                     {
-                        info.SqlConnection.ChangeDatabase(newDatabaseName);
+                        if (connection.State == ConnectionState.Open)
```

Comment:
That is a good question. I moved this line, so it is marked as an addition by git, but this is the current behavior of the service.